#!/usr/bin/env python
PACKAGE = "path_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("planner_geotiff_map", str_t, 0, "Path to GeoTIFF map for planner", "")
gen.add("non_coverage_turning_radius", double_t, 0, "Turning radius of the vessel (meters)", 8, 1, 100)
gen.add("coverage_turning_radius", double_t, 0, "Turning radius acceptable during sonar operation (meters)", 16, 1, 100)
gen.add("max_speed", double_t, 0, "Maximum speed of the vessel (meters/second)", 2.5, 0, 30)
gen.add("line_width", double_t, 0, "Acceptable across-track distance from survey line (m)", 2, 0.05, 10)
gen.add("branching_factor", int_t, 0, "Branching factor for connecting samples (doubled for coverage/non-coverage modes", 9, 1, 50)
gen.add("dump_visualization", bool_t, 0, "Toggle visualization info dump", False)
gen.add("visualization_file", str_t, 0, "Visualization file", "/tmp/planner_visualization")

exit(gen.generate(PACKAGE, "path_planner", "path_planner"))
